### ISSUE TITLE
Prometheus manifest: Weird RBAC/NameSpace/ServiceAccount setup

### ISSUE THREAD

**Issue by Ornias1993** [metallb/metallb#1690]
The current prometheus compatible manifest is a tad weird:
A. It requires the use of the "monitoring" namespace for prometheus-operator, which isn't actually a standard/default by prometheus
B. It requires a serviceAccount named `prometheus-k8s` which isn't actually... created or used by prometheus-operator
C. Its only used to create the RBAC setup, is not actually used/referenced anywhere in the manifest
D. ServiceMonitors don't actually need anything byond... the servicemonitor itself, so why the whole RBAC/NameSpace/serviceAccount-overcomplication is there, is a mystery to me.

TLDR:
I could setup monitoring myself with just a servicemonitor, so why is the while namespace/serviceaccount/rbac binding thing there at all? To be this looks like someone many years ago used created this and no one bothered to review the design *or* I might be severely overlooking something...

**Comment by fedepaol** [comment]
Hi! Thanks for raising this!

First things first: the all-in-one manifests are, as their name suggests, all-in-one. For this reason, they do not offer the same level of configurability that both the helm chart and the kustomize based setup offer.

This particular set of manifests (the prometheus ones) is based on a deployment of the prometheus-operator that comes from the official prometheus operator's [kube prometheus repo](https://github.com/prometheus-operator/kube-prometheus), and serves two purposes. The first one is to offer a possible configuration to work with a deployment of the prometheus operator, and the second one is to deploy that particular setup in our CI, to make sure that the metrics we think we are exposing are exposed.


Let's talk about the setup then:
This configuration is based upon a customization of https://github.com/prometheus-operator/kube-prometheus that can be found [here](https://github.com/metallb/metallb/tree/main/dev-env/kube-prometheus) where what we did is to strip away some bits to be lighter on our CI environment.

A/B - The `monitoring` namespace and the `prometheus-k8s` SA come from the kube-prometheus deployment
C - it is used to give the prometheus instance the permissions to scrape the metallb-system namespace for endpoints, services and what not because without them, the prometheus operator won't be able to read them
D - this really depends on how the prometheus operator is deployed and how many permissions it gets by default. kube-prometheus tries to be low on proviledges (see the discussion in https://github.com/prometheus-operator/kube-prometheus/issues/350) but offers also a way to override. 

Unfortunately there's no one-size-fit-all solution we can apply to the manifests. What we can do is to be more clear in documenting that this is A prometheus-operator configuration that could not fit all the deployments.

Hope this helps

**Comment by Ornias1993** [comment]
So TLDR:
The metrics version is designed to work with `kube-prometheus` manifest and not designed to work with the stock `prometheus-operator` manifest .

However, this serieusly important piece of information is not in the documenation as far as i'm aware?

**Comment by fedepaol** [comment]
> So TLDR: The metrics version is designed to work with `kube-prometheus` manifest and not designed to work with the stock `prometheus-operator` manifest .
> 

Yep, that's the idea.

> However, this serieusly important piece of information is not in the documenation as far as i'm aware?

The documentation does not even mention those manifests https://metallb.universe.tf/installation/ so they are not meant to deploy metallb (not officially, at least).

Closing this issue, will file another one to expand the docs. Thanks again for raising this!

**Comment by Ornias1993** [comment]
Whoops forgot to add:
Yes that closes this issue :)

The docs do mention it:
"There are also two all-in-one manifests to allow the integration with prometheus. They assume that the prometheus operator is deployed in the monitoring namespace using the prometheus-k8s service account. It is suggested to use either the charts or kustomize if they need to be changed."

Important to note the following:
"They assume that the prometheus operator is deployed..."

The primary issue with this is the serviceAccount, which is not even an operator released object.

But I think you very well get the jizz by now, thanks anyway @fedepaol 👍
